pydantic/dataclasses.py

```diff
@@ -253,10 +248,7 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         cls.__pydantic_complete__ = pydantic_complete  # type: ignore
         return cls
 
-    if _cls is None:
-        return create_dataclass
-
-    return create_dataclass(_cls)
+    return create_dataclass if _cls is None else create_dataclass(_cls)
```

Comment:
Imho I think I prefer the current form:

```python
if _cls is None:
    return create_dataclass

return create_dataclass(_cls)
```

As it makes it easier to understand the two cases, when the decorator is applied as is (`@dataclass`) and with arguments (`@datalcass(...)`).

But feel free to ignore, this is very minor

Reply:
I think I prefer the one liner here, but thanks for the feedback!

Reply:
One nice thing about this new approach is that it removes the need for the noqa above